src/ol/typedefs.js

```diff
@@ -632,39 +605,30 @@ ol.interaction.SnapSegmentDataType;
  * A projection as {@link ol.proj.Projection}, SRS identifier string or
  * undefined.
  * @typedef {ol.proj.Projection|string|undefined} ol.proj.ProjectionLike
- * @api stable
  */
 ol.proj.ProjectionLike;
 
 
 /**
  * A function that takes an array of input data, performs some operation, and
- * returns an array of ouput data.  For `'pixel'` type operations, functions
- * will be called with an array of {@link ol.raster.Pixel} data and should
- * return an array of the same.  For `'image'` type operations, functions will
- * be called with an array of {@link ImageData
- * https://developer.mozilla.org/en-US/docs/Web/API/ImageData} and should return
- * an array of the same.  The operations are called with a second "data"
- * argument, which can be used for storage.  The data object is accessible
- * from raster events, where it can be initialized in "beforeoperations" and
- * accessed again in "afteroperations".
+ * returns an array of ouput data.
+ * For `pixel` type operations, the function will be called with an array of
+ * pixels, where each pixel is an array of four numbers (`[r, g, b, a]`) in the
+ * range of 0 - 255. It should return an array of the same.
+ * For `'image'` type operations, functions will be called with an array of
+ * {@link ImageData https://developer.mozilla.org/en-US/docs/Web/API/ImageData}
+ * and should return an array of the same.  The operations are called with a
+ * second "data" argument, which can be used for storage.  The data object is
+ * accessible from raster events, where it can be initialized in
+ * "beforeoperations" and accessed again in "afteroperations".
  *
- * @typedef {function((Array.<ol.raster.Pixel>|Array.<ImageData>), Object):
- *     (Array.<ol.raster.Pixel>|Array.<ImageData>)}
- * @api
+ * @typedef {function((Array.<Array.<number>>|Array.<ImageData>), Object):
+ *     (Array.<Array.<number>>|Array.<ImageData>)}
```

Comment:
It's clear that the compiler isn't helping us here.  This should be:
```
function((Array<Array<number>>|Array<ImageData>), Object): (Array<number>|ImageData)
```

That is, a raster operation takes either an array of pixels or an array of `ImageData` and returns either a single pixel or a single `ImageData` object.  For a `pixel` type operation, the function takes an array of pixels and returns a single pixel.  For an `image` type operation, the function takes an array of `ImageData` and returns a single `ImageData` object.